### PR TITLE
11 event qeue

### PR DESCRIPTION
**Pull request by sinelanguage**
connects to rangle/angular-training-slides#43

### CODE REVIEW

File: language-examples/remedial-js/11.0-event-qeue/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+Running applications in a web browser involves a collection of technologies, such as a JavaScript Engine (such as Chrome’s V8), a collection of Web API’s (such as the DOM), and the Event Loop and Event Queue.
+
+### What exactly does the JavaScript Engine do?
+
+It’s job is to go through all the lines of JavaScript in an application and process them one at a time. This means that JavaScript is single-threaded. The main repercussion of this is that if you’re running a line of JavaScript that happens to take a long time to return, then all the code after that will be blocked.
+
+**So how does javascript engine know how to process a single line of javascript at a time?**
+
+It uses a call stack. You can think of a call stack like entering an elevator — the first person who enters the elevator is the last person to exit the elevator, whereas the last person to enter is the first to exit.
```
Comment (turbobeast) [inline language-examples/remedial-js/11.0-event-qeue/README.md]: You've provided a metaphor for the call stack, but failed to mention that it is the stack of functions being executed at a given time.

File: language-examples/remedial-js/11.0-event-qeue/app/exercise.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+'use strict';
+'format es6'; // force SystemJS to transpile exercise
+
+const exampleDiv = document.getElementById('Please see Readme file for full details about the Event Qeue');
```
Comment (turbobeast) [inline language-examples/remedial-js/11.0-event-qeue/app/exercise.js]: this is a crazy `id` for an element!
did you mean to write..?
```javascript
document.getElementById('example')
  .appendChild(document.createTextNode('Please see Readme file for full details about the Event Queue'))
```

File: language-examples/remedial-js/11.0-event-qeue/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+### Where does the event qeue and loop play into this?
+
+The way Javascript avoids blocking code is that the engine provides a mechanism and it’s via asynchronous callback functions.
+
+An asynchronous callback function is just like any other function you’re used to writing in JavaScript, with the added caveat that it doesn’t get executed till later.
+
+### How is a function posted to the event qeue?
+
+When an async function is executed (like setTimeout), something special happens to its callback — the browser places the callback function into an Event Table. Think of the event table as a registration booth: the call stack tells the event table to register a particular function to be executed only when a specific event happens.
```
Comment (turbobeast) [inline language-examples/remedial-js/11.0-event-qeue/README.md]: the wording of this is strange. Apart from ES7 async functions, I don't think there is a such thing as an async function in javascript, just async APIs (`requestAnimationFrame`, `XMLHttpRequest`, `setTimeout`...). `setTimeout()` executes synchronously and pushes another completely synchronous function onto the event queue.

File: language-examples/remedial-js/11.0-event-qeue/app/exercise.js (hunk 2 above)
Comment (sinelanguage) [reply language-examples/remedial-js/11.0-event-qeue/app/exercise.js]: yes :) thank you

Comment (sinelanguage) [reply language-examples/remedial-js/11.0-event-qeue/app/exercise.js]: just did element.innerHTML to match the other files

File: language-examples/remedial-js/11.0-event-qeue/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+Lets walk through this:
+
+1. The call to greet causes control(the JS engine) to jump to the start of that function (line 2).
+2. It calls console.log (a built-in browser function), which takes control, does its job, and then returns control to line 2.
+3. Then it reaches the end of the greet function, so it returns to the place that called it, at line 4.
+4. The line after that calls console.log again.
+
+Because a function has to jump back to the place of the call when it returns, the computer must remember the **context** from which the function was called. In one case, console.log has to jump back to the greet function. In the other case, it jumps back to the end of the program.
+
+The place where the computer stores this context is the call stack. Every time a function is called, the current context is put on top of this “stack”. When the function returns, it removes the top context from the stack and uses it to continue execution.
```
Comment (turbobeast) [inline language-examples/remedial-js/11.0-event-qeue/README.md]: 👍

File: language-examples/remedial-js/11.0-event-qeue/README.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
+
+# 11.0 - Example Event Qeue
```
Comment (turbobeast) [inline language-examples/remedial-js/11.0-event-qeue/README.md]: `Qeue` should be `Queue` ...everywhere

File: language-examples/remedial-js/11.0-event-qeue/README.md (hunk 1 above)
Comment (ggprod) [reply language-examples/remedial-js/11.0-event-qeue/README.md]: added that call-stack definition

File: language-examples/remedial-js/11.0-event-qeue/README.md (hunk 5 above)
Comment (ggprod) [reply language-examples/remedial-js/11.0-event-qeue/README.md]: fixed

**Pull request merged**